### ISSUE TITLE
Problem when both rooks or both knights can move to the same square

### ISSUE THREAD

**Issue by Jeptah** [Sentero-esp12/Multi-Premoves-Mouse-Keyboard#16]
I'll talk about rooks but it applies to knights also. When both rooks can move to a square, and you hover the square and press the key of the right rook, it doesn't move. Pressing the key of the left rook sometimes moves the left rook and sometimes the right rook.

**Comment by jfaure** [comment]
Have you considered simply playing both moves one after another ? Seems like a very simple solution.

if the first move is illegal lichess will ignore it and play the second
otherwise the second move will be illegal

**Comment by Jeptah** [comment]
Both moves are legal, but you can only move the left piece. It's a bug.

**Comment by Sentero-esp12** [comment]
I'm trying to reproduce but haven't been able to yet. What keys are you using for rooks? In what positions? Does it happen each time or occasionally?

**Comment by Jeptah** [comment]
With the new one I'm getting an error: "Unrecognized manifest key 'update'." 
![multi-premove-error](https://user-images.githubusercontent.com/72153982/96336050-90257200-1085-11eb-89dc-add6313553e8.png)

**Comment by Sentero-esp12** [comment]
It's just a warning, because I store information in the manifest and Chromium thinks it's not supposed to be there, just ignore it. Is it preventing the installation, by the way?

**Comment by Jeptah** [comment]
I don't think it prevents installation, but it doesn't seem to work. I get a few more errors.

![multi-premove-error](https://user-images.githubusercontent.com/72153982/96336348-c106a680-1087-11eb-8f51-911dfe29fb1e.png)

**Comment by Sentero-esp12** [comment]
Thanks for the info, so it doesn't work with unlimited time control. Could you try a limited time control?

As for the rooks and knights, it works properly for me. I guess the only way to find out why it's happening to you is via Teamviewer.

**Comment by Sentero-esp12** [comment]
Unlimited time control issue should probably be fixed.

**Comment by Jeptah** [comment]
Okay, sorry for only checking in unlimited time. On timed game, I can make moves and everything works, except for when both rooks can move to same square.

https://lichess.org/yUoCo67uqp3c
On move 5, I pressed '1'. Right rook moved.

**Comment by Sentero-esp12** [comment]
Could you take a screenshot of your settings?

**Comment by Jeptah** [comment]
![Uploading multi-premove-error.png…]()

**Comment by Sentero-esp12** [comment]
Thanks. Did you reload the page during the games where this bug happened?

**Comment by Jeptah** [comment]
Without any reload during game. I just saw that after reload the right/left rooks can switch, but that's not the source of the bug.

**Comment by Sentero-esp12** [comment]
I see, thanks for the info. If you are fine with trying Teamviewer, I could check what exactly causes the bug, because I cannot reproduce it.

**Comment by jfaure** [comment]
I have the same problem when rooks/knights can reach a square, on commit 583b21c.
Unfortunately if I update the script, keyboard inputs are completely ignored.

I'm happy to help on discord j4#0303

**Comment by Sentero-esp12** [comment]
Could you try reinstalling with the latest update, it should probably be fixed.

**Comment by jfaure** [comment]
Latest branch still ignores keyboard for me, any idea why ? I'm using google-chrome on a linux system.

cherry-picking the most recent commit atop 583b21c gives slightly better results, however
if knight conflict: only left knight button works, and moves the wrong knight
if rook conflict: only left rook works if ambiguous, and which rook is actually moved seems random (possibly the closest to dest square)

**Comment by Sentero-esp12** [comment]
`Latest branch still ignores keyboard for me, any idea why ? I'm using google-chrome on a linux system.`
Did you enable it in the settings? 

Yeah, the knights and rooks should only be fixed in the latest update.

**Comment by jfaure** [comment]
enable what exactly ? the ultrabullet theme work so I'm assuming the extension is properly loaded

I have the latest update, I only skipped some in the middle.

**Comment by Sentero-esp12** [comment]
Also is it working for the mouse? Is it showing the circle under the board?

**Comment by jfaure** [comment]
There is no circle
I've found the problematic commit: bc643ef79d927c23654ddcb089bb4159eab9ef2f

**Comment by Sentero-esp12** [comment]
I've just tested the extension on Linux in Chromium and it worked well for me. Try completely removing the folder, removing Lichess cookies and downloading again from Github. Maybe there is a problem with local storage/files.

**Comment by jfaure** [comment]
Same thing, no circle
have you tried chrome ?

**Comment by Sentero-esp12** [comment]
I don't have Chrome there, so I won't be able to test it any time soon, but I guess it should be the same

**Comment by jfaure** [comment]
Well keyboard works on chromium, but with the same issues for conflicting pieces. bc643ef seems to have broken chrome somehow.

**Comment by Sentero-esp12** [comment]
I've just installed Chrome on Linux and it works there as well for me, and no issue with rooks/knights

**Comment by thecodingterm** [comment]
> I used '1' and '3' fore left and right rook, and 'a' and 'd' for left and right knight. It would happen each time in any position where both left and right pieces can make the move.
> 
> With the new one I'm getting an error: "Unrecognized manifest key 'update'."
> ![multi-premove-error](https://user-images.githubusercontent.com/72153982/96336050-90257200-1085-11eb-89dc-add6313553e8.png)

Im getting the same thing today @Sentero-esp12 maybe can you like an older october version?

**Comment by Sentero-esp12** [comment]
Cannot reproduce. If someone with this issue wants to have a Teamviewer/screen share session, write here, so I reopen this issue.

**Comment by Sentero-esp12** [comment]
Can reproduce, happens all the time.

**Comment by Sentero-esp12** [comment]
This is related to zoom. At 100% this bug doesn't occur. I'll try to fix it. Meanwhile you can set your page zoom to 100% to fix this.

**Comment by Sentero-esp12** [comment]
I think this should be fixed. Write here if you still experience this bug in the latest version.

**Comment by Sentero-esp12** [comment]
This still happens
